File: py/branch.py
#!/usr/bin/env python3.7
'''branch.py'''

from time import time_ns

NUM_OF_LOOPS = 1000000000


def main():
    start = time_ns()
    for i in range(0, NUM_OF_LOOPS):
        if i % 1:
            pass
        if i % 2:
            pass
        if i % 3:
            pass
        if i % 4:
            pass
        if i % 5:
            pass
        if i % 6:
            pass
        if i % 7:
            pass
        if i % 8:
            pass
        if i % 9:
            pass
        if i % 10:
            pass
    end = time_ns()
    print( (end-start) / 1000000000)


if __name__ == '__main__':
    main()

File: pl/branch.pl
#!/usr/bin/env perl
# branch.pl

use Time::HiRes;

my $NUM_OF_LOOPS = 1000000;

# Gather the start time of the loops
my $start = Time::HiRes::gettimeofday();

# Loop through an empty loop 1 billion times
for (my $i=0; $i < $NUM_OF_LOOPS; $i++) {
    if ($i == 1) {}
    if ($i == 2) {}
    if ($i == 3) {}
    if ($i == 4) {}
    if ($i == 5) {}
    if ($i == 6) {}
    if ($i == 7) {}
    if ($i == 8) {}
    if ($i == 9) {}
    if ($i == 10) {}
}
# Gather the end time of the loops
my $end = Time::HiRes::gettimeofday();

# Print the amount of time taken
printf("%.10f\n", $end - $start);
